make sure to serve the axon first

camfairchild opened the pull request:
This is a hotfix for #120 
This change serves the axon to the chain

opentaco reviewed (commented):
A `--neuron.axon_off` flag would be useful, to make axon serving optional, unless there is a reason the axon is mandatory.

This should probably require a manual update, so that validators can specify their `--axon.port --axon.external_port --axon.external_ip` settings correctly, given that axon is only now introduced.

camfairchild commented:
> A `--neuron.axon_off` flag would be useful, to make axon serving optional, unless there is a reason the axon is mandatory.

We made it mandatory because of a DoS attack. If your Validator has no IP served, it is blacklisted in the firewall of most miners

camfairchild commented:
> A `--neuron.axon_off` flag would be useful, to make axon serving optional, unless there is a reason the axon is mandatory.
> 
> This should probably require a manual update, so that validators can specify their `--axon.port --axon.external_port --axon.external_ip` settings correctly, given that axon is only now introduced.

Added your suggestion in [eb39381](https://github.com/opentensor/validators/pull/127/commits/eb393815e205da6a907391650ce7fe4c381db610)

Also added a catch around the process to log without crashing if it fails [99f64b8](https://github.com/opentensor/validators/pull/127/commits/99f64b84aee2ccd2c4dc09bf7e735b7c48dc3438)

opentaco commented on openvalidators/config.py:
```diff
@@ -166,6 +166,14 @@ def add_args(cls, parser):
         default=4096,
     )
 
```
A comment in the code here would be useful to explain why the axon should be on, and warn that turning it off will e.g. risk being blacklisted.

opentaco commented on openvalidators/config.py:
```diff
@@ -166,6 +166,14 @@ def add_args(cls, parser):
     )
 
+    parser.add_argument(
+        "--neuron.axon_off",
```
A comment in the code here would be useful to explain why the axon should be on, and warn that turning it off will e.g. risk being blacklisted.
